dbms/src/Interpreters/convertFieldToType.cpp

```diff
@@ -181,8 +181,13 @@ Field convertFieldToTypeImpl(const Field & src, const IDataType & type, const ID
         if (!which_type.isDateOrDateTime() && !which_type.isUUID() && !which_type.isEnum())
             throw Exception{"Logical error: unknown numeric type " + type.getName(), ErrorCodes::LOGICAL_ERROR};
 
-        /// Numeric values for Enums should not be used directly in IN section
-        if (src.getType() == Field::Types::UInt64 && !which_type.isEnum())
+        if (which_type.isEnum() && (src.getType() == Field::Types::UInt64 || src.getType() == Field::Types::Int64))
+        {
+            /// Convert UInt64 or Int64 to Enum's value
+            return dynamic_cast<const IDataTypeEnum &>(type).castToValue(src);
+        }
+
+        if (src.getType() == Field::Types::UInt64)
```

Comment:
Why do we need this branch?

Reply:
Will try to investigate. Seems like my changes doesn't really help.

Reply:
UPD: It does work the same as String now.
I was surprised with
```
CREATE TABLE enums (x Enum8('hello' = 0, 'world' = 1), y String) engine = Memory;
insert into enums values ('hello', 'find me'), ('hello', 'and me');
select y in ('hello') from enums;
┌─in(y, 'hello')─┐
│              0 │
│              0 │
└────────────────┘
select y in ('world') from enums;
┌─in(y, 'world')─┐
│              0 │
│              0 │
└────────────────┘

```

Reply:
Ok. And the only remaining question: do we still need this branch?

Reply:
The only problem i have found is in table function values :). When we do `const TupleBackend & value_tuple = value_field.safeGet<Tuple>().toUnderType();` to iterate the tuple values. All the fileds also go to under type and Data becomes UInt64.

Reply:
I believe that branch is not needed anymore. But not sure 100%. Deleting It also helped me to find bug in table function values.